### DISCUSSION TITLE
Out-sourcing development on the Bisq mobile app?

### DISCUSSION THREAD

**Discussion by devinbileck** [bisq-network/bisq#6108] [General]
I am curious if anyone has any opinions or ideas on out-sourcing development on the [Bisq mobile app](https://github.com/bisq-network/bisq-client)?

I have not been able to find any contributors to help nor have I been able to make much meaningful progress myself, mainly due to a lack of time and inexperience with KMP. As a result, out of curiosity I reached out to a couple reputable company's that seem up to the task. I haven't committed to anything, but now have a basic idea how they work and how much they would charge for a fixed period of time. Though I don't yet know how much time it would take for certain functionality to be implemented or to deliver an MVP. Ideally, I would like to deliver all milestones described in the [project plan](https://github.com/bisq-network/bisq-client/wiki/Project-Plan), but would need further discussions with them to work out a cost and time frame.

Even if we were to just get the project off the ground with a basic foundation and functioning MVP, it should be a catalyst to help jump start the project and might help attract contributors especially inexperienced contributors as they would have a foundation to build from.

I don't mind driving this project forward, and before I commit to anything I would first work out more details and lay out a time frame with costs and deliverables to ensure it would deliver value and remain within our dev budget.

However, I would need to secure funding up front. I don't think I will be able to source all the funds myself and would need to find alternatives, possibly crowdfunding or a joint contribution.

Just thought I would open this idea up for discussion in case anyone has any opinions or ideas...

**Comment by ripcurlx** [comment]
Hi @devinbileck ! I completely missed this discussion. Outsourcing it, might be an option, but has also multiple drawbacks as you are dependent on a codebase of someone that might only wanted to get the thing done without the required quality, someone else wants to build upon. I'll slowly having more time for Bisq again and can have a look as well, if we are able to do some groundwork ourselves.

**Comment by devinbileck** [comment]
Thanks for the reply @ripcurlx.

I think having well defined requirements, short delivery cycles (for which the deliverables are reviewed), and a reputable company would help mitigate most of those concerns. Besides, one company that I am considering wrote the book on clean code ;)

In any case, I have put this idea on the back burner for now until I figure a way to obtain funding.

In the meantime, I have been spending more development time recently. My focus has been getting the existing apps working as there are numerous outstanding issues. I just published an update with a couple fixes for the iOS app, and am wrapping up a major overhaul of the Android app and should be publishing an update soon.

After that, I will focus my time on the new app and see if I can get something working.

**Reply by ripcurlx** [reply]
That sounds great! As I also started tinkering with a cross-platform mobile setup it would be great if we could arrange a quick call. I DM you on Matrix.

I played around with https://github.com/square/workflow which is quite similar to what I used in other client-side technologies and am trying out right now how mature KMM is already for production use.
